theano/gpuarray/extra_ops.py

```diff
@@ -450,12 +450,13 @@ def c_support_code_struct(self, node, nodename):
         """ % locals()
         return super(GpuCumsum, self).c_support_code_struct(node, nodename) + code
 
-
+@register_opt('fast_compile')
 @op_lifter([CumsumOp])
-def use_gpu_cumsumop(node, ctx_name):
-    if node.inputs[0].dtype == 'float32':
-        axis = node.op.axis
-        x = node.inputs[0]
+@register_opt2([CumsumOp], 'fast_compile')
```

Comment:
It is this line that cause the error:

468  ->         x = GpuFromHost(ctx_name)(x)

Use as_gpuarray_variable(x, ctx_name) instead.

On Thu, Jun 9, 2016 at 9:05 AM, Ramana Subramanyam <notifications@github.com
> wrote:

> In theano/gpuarray/extra_ops.py
> <https://github.com/Theano/Theano/pull/4570#discussion_r66436036>:
>
> >  @op_lifter([CumsumOp])
> > -def use_gpu_cumsumop(node, ctx_name):
> > -    if node.inputs[0].dtype == 'float32':
> > -        axis = node.op.axis
> > -        x = node.inputs[0]
> > +@register_opt2([CumsumOp], 'fast_compile')
>
> (Sorry, I had pasted the wrong command in my previous comment.)
>
> I had just tested the latter comment. Code remains the way it is in the
> first comment. For reproducing,
> THEANO_FLAGS=device=cuda,lib.cnmem=0.6 theano-nose
> theano/gpuarray/tests/test_extra_ops.py
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4570/files/abc2020d8f87a5ca3fa31601604a2dffac3709fd#r66436036>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALC-4jNK3yBbv-npNywcv_b_Sg_4icyks5qKA9_gaJpZM4Ir4rI>
> .
>